engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -80,7 +82,14 @@ final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionSt
     this.copy(statusStore = statusStore ++ values, hasNewRunnables = hasNewRunnables || values.values.exists(_.isTerminal))
   }
 
-  def runnableScopes = keysWithStatus(NotStarted) filter arePrerequisitesDone
+  /**
+    * Returns the list of jobs ready to be run, along with a Boolean indicating whether or not the list has been truncated.
+    * The size of the list will be MaxJobsToStartPerTick at most. If more jobs where found runnable, the boolean will be true, otherwise false.
+    */
+  def runnableScopes: (List[JobKey], Boolean) = {
+    val readyToStart = keysWithStatus(NotStarted).toStream filter arePrerequisitesDone
+    (readyToStart.take(ExecutionStore.MaxJobsToStartPerTick).toList, readyToStart.size > ExecutionStore.MaxJobsToStartPerTick)
```

Comment:
Note to self: fix this: readyToStart.size evaluates the whole stream which negates the reason of making it a stream entirely.